addons/mail/models/mail_activity.py

```diff
@@ -577,6 +577,13 @@ def action_open_document(self):
             'view_mode': 'form',
         }
 
+    def action_snooze(self):
+        today = date.today()
+        for activity in self:
+            activity.write({
+                'date_deadline': timedelta(days=7) + max(activity.date_deadline, today)
+            })
```

Comment:
:socks: Could use an update instead of a write, and I would make the base argument (date) first, easier to read in my opinion (date + snooze duration)
```suggestion
            activity.date_deadline = max(activity.date_deadline, today) + timedelta(days=7)
            })
```